NimbleDroid/NimbleDroid.swift

```diff
@@ -7,12 +7,46 @@
 //
 
 public class NDScenario : NSObject {
+    static var coldStartup = false
+    static var initialized = false
+    static var scenario : String? = nil
+
+    public class func setup() {
+        if initialized {
+            NSLog("NDScenario already initialized")
+            return
+        }
+        let processInfo = ProcessInfo.processInfo
+        let arguments = processInfo.arguments
+        let processIdentifier = processInfo.processIdentifier
+        if let scenarioIndex = arguments.index(of: "--ndScenario") {
+            if let bookendIndex = arguments.index(scenarioIndex, offsetBy: 1, limitedBy: arguments.endIndex - 1) {
+                scenario = arguments[bookendIndex]
+                NSLog("NDScenario initialized pid %d scenario %@", processIdentifier, scenario!)
+            } else {
+                NSLog("NDScenario failed to initialize pid %d, --ndScenario missing bookendID", processIdentifier)
+            }
+        } else if arguments.contains("--ndColdStartup") {
+            coldStartup = true
+            NSLog("NDScenario initialized pid %d coldStartup", processIdentifier)
+        } else {
+            NSLog("NDScenario initialized pid %d", processIdentifier)
+        }
+        initialized = true
+    }
+
     public class func begin(bookendID : String) {
+        if bookendID == scenario {
```

Comment:
Can we add some debug log if it's used w/o setup()?

Reply:
K, go ahead w/ what you think good. (But in general, logs are not bad even if they are redundant.)

Reply:
Now that I think about it, if the user is testing the framework locally this debug message will let them know they've forgotten to initialize. It's not redundant after all :)